### ISSUE TITLE
508-defect-2 [HEADINGS]: Headings should be logically ordered on the VA.gov homepage

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#27976]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I expect headings to be logically ordered so I can understand the hierarchy of a page and navigate efficiently.

## Details

Several headings on the VA.gov homepage are not logically ordered which can create misleading groupings on the page.
- The `h1` does not contain any information that this is the homepage of VA.gov; for folks using a screen reader who navigate back to the `h1` to determine what page they are on, it won't be clear to them that this is the homepage
- Duplicate `h2`s exist like disability and health care
- Many sections lack headings which can create misleading hierarchies (see example below)

<img width="1268" alt="Screen Shot 2021-07-28 at 9 25 20 AM" src="https://user-images.githubusercontent.com/14154792/127330658-990bcb95-9cb5-4014-b068-89633ce77bd4.png">

## Acceptance Criteria

- [ ] Headings are logically ordered
- [ ] The `h1` contains part of the `title` element and makes clear that this is the homepage of VA.gov

## Steps to Recreate

Use the chrome extension headingsMap and read it like a table of contents. If it does not make logical sense, headings should be added or reorganized.

## Proposed Solution

Currently, the home page's headings look like this if read as a table of contents:

1. Access and manage your VA benefits and health care (h1)
  1. Health care (h2)
  2. Disability
  3. Education
  4. Records
  5. Service Member Records
  6. Family Member Benefits
  7. Burials and Memorials
  8. Careers and Employment
  9. Housing Assistance
  10. Pension
  11. Life Insurance
  12. Education and Training
  13. Records
  14. Health Care
  15. Disability 
    1. VA trust report (h3)
    2. Celebrating 75 years
    3. VAntage Point
  16. Veteran Programs and Services
  17. More VA Resources
  18. Get VA Updates
  19. In Crisis? Talk to Someone Now
  20. Get Answers
  21. Call Us
  22. Visit a medical center
  23. Language assistance

It would be much easier to understand if it were organized as:

1. Welcome to VA.gov (h1)
  1. Popular on VA.gov (h2)
    1. Health care (h3)
    2. Disability
    3. Education
    4. Records
  2. Services and Information (h2)
    1. Service Member Records (h3)
    2. Family Member Benefits
    3. Burials and Memorials
    4. Careers and Employment
    5. Housing Assistance
    6. Pension
    7. Life Insurance
    8. Education and Training
    9. Records
    10. Health Care
    11. Disability 
  3. Tools and Resources (h2)
    (This would house links to the facility locator, talk to Veterans crisis line, etc button links)
  3. Latest News (h2)
    1. VA trust report (h3)
    2. Celebrating 75 years
    3. VAntage Point
  4. Footer (h2)
    1. Veteran Programs and Services (h3)
    2. More VA Resources
    3. Get VA Updates
    4. In Crisis? Talk to Someone Now
    5. Get Answers
    6. Call Us
    7. Visit a medical center
    8. Language assistance

Any newly added headings should be visible to screen reader and sighted users rather than `sr-only` elements unless there is a strong design reason otherwise. 

## WCAG or Vendor Guidance (optional)

[Common A11y Issues on VA.gov: Headings](https://docs.google.com/presentation/d/1xBentIqwKrcIu2VzAkws3mp6E2Gny1EkmkaS_VkPS14/edit?usp=sharing)

**Comment by joshkimux** [comment]
@brianalloyd and @kelsonic , during a recent audit while I was making a presentation on common a11y issues, I noticed that the headings on VA.gov may be creating barriers to access for our screen reader users. This also came up during in research during screen reader usability tests, so this is something I'd recommend prioritizing to fix-- especially since there are usability studies in the pipeline where screen reader users may be attempting to access the homepage.

Are y'all the right team to forward this too, or should we be including others as well? Would love y'alls help as always 🙇

**Comment by kelsonic** [comment]
We can definitely help with this 🙂 It might require some coordination with the CMS team (content editors specifically) to rearrange the content a bit, but we should be able to take care of most of the changes in the homepage templates

**Comment by brianalloyd** [comment]
Thanks @joshkimux as @kelsonic mentioned we'll add this to our backlog and work it into our burn down. Appreciate it.

**Comment by joshkimux** [comment]
@brianalloyd and @kelsonic y'all are the best. Thank you so much!

**Comment by cindymerrill** [comment]
@kelsonic I don't think this work will require rearranging the content but instead just adding or changing HTML headings to mirror the visual structure of the home page. Does that kind of change require the involvement of content editors?

**Comment by kelsonic** [comment]
Probably not 🙂

**Comment by cindymerrill** [comment]
Then @joshkimux knows exactly what needs fixing because he gave a really good presentation to the UX team about it.

**Comment by joshkimux** [comment]
@kelsonic let's connect on this! We can use `sr-only` classes as an intermediary fix that won't change the existing visual design/content.

**Comment by davidconlon** [comment]
@brianalloyd before we pull the trigger on this I need to understand how much refactoring this will take. As I mentioned in other threads we are looking at a full rebuild of the homepage and this page has been in production with this structure for over 3 years so need to think through ROI.

**Comment by joshkimux** [comment]
Attaching some screenshots so it's a little more clear what changes are being proposed visually.

## Option 1: Visible headings
Ideally, headings would be visible to help folks contextualize sections. There's strong a11y, content, and SEO benefits from having well structured headings.

![headings-visual](https://user-images.githubusercontent.com/14154792/140517796-6662dd73-6d4c-464a-a6b8-220d5125e516.png)

## Option 2: Screen reader only headings
Alternatively, if we want a less intrusive change (visually) while keeping the a11y, content, and SEO benefits this ticket provides; we can also consider using `sr-only` classes as a temporary fix.

![No changes](https://user-images.githubusercontent.com/14154792/140517841-d87b8bad-b49d-48f8-8714-824eb3be05f9.png)

**Comment by cindymerrill** [comment]
Either of these options would greatly improve the navigation experience for Veterans using screen readers because it would enable them to "see" the structure of the page and thus better understand the content links and their groupings @davidconlon @joshkimux

**Comment by jimuhh** [comment]
I agree with @cindymerrill. Re-structuring the content to make it easier for vision impaired users to keep the navigation structure in their heads is a usability/IA win for everyone. A great example of  [curb cut effect ](uhttps://uxdesign.cc/the-curb-cut-effect-and-universal-product-design-964be1c6c841#:~:text=Disability%20activists%20and,we%20forget%20its%20original%20purpose.)where designing for disability empowers everyone.
